src/parpeloadbalancer/loadBalancerMaster.cpp

```diff
@@ -32,13 +32,11 @@ void LoadBalancerMaster::run() {
     // have to initialize before can wait!
     sendRequests.resize(numWorkers, MPI_REQUEST_NULL);
 
-    /* Create semaphore to limit queue length
-     * and avoid huge memory allocation for all send and receive buffers.
-     */
-    unsigned int queueMaxLength = mpiCommSize;
+    // Create semaphore to limit queue length
 #ifdef SEM_VALUE_MAX
-    if(SEM_VALUE_MAX < queueMaxLength)
-        throw ParPEException("SEM_VALUE_MAX too small to work with the given MPI_Comm_size.");
+    unsigned int queueMaxLength = SEM_VALUE_MAX;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Skipping configuration 'PARPE_ENABLE_MPI;SEM_VALUE_MAX' since the value of 'SEM_VALUE_MAX' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.](https://app.codacy.com/manual/dweindl/parPE/pullRequest?prid=4904108)